### PR TITLE
[FIX] web: Datepicker uses static date format

### PR DESCRIPTION
**Pull request by Arcasias**
The current datepicker implementation uses Luxon to send and receive
dates from the client and Moment to create and communicate with the
bootstrap datepicker.

Before this commit: the current database date/time format was used to
convert dates from one system to the other. However there are
inconsistencies in notation for some formats, for which the conversion
could not be made properly.

Now: a static format is used to translate dates and the actual database
format is applied only after both systems have been notified of a date
change.

Tasks [2794506](https://www.odoo.com/web#id=2794506&cids=1&menu_id=5879&action=3531&model=project.task&view_type=form), [2797949](https://www.odoo.com/web#id=2797949&menu_id=5879&action=3531&model=project.task&view_type=form&cids=1)

### CODE REVIEW

**Comment by adr-odoo** [comment]
@pparidans

File: addons/web/static/src/core/datepicker/datepicker.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const { Component, hooks } = owl;
 const { DateTime } = luxon;
 const { useExternalListener, useRef, useState } = hooks;
 
+const now = DateTime.local();
```
Comment (aab-odoo) [inline addons/web/static/src/core/datepicker/datepicker.js:11]: I'd move this next to `isValidStaticFormat`, and maybe name it something else than `now` (it is indeed _now_ when you create it, but it's no longer _now_ later on (weird to save a "now" value).

**Review by aab-odoo — APPROVED** [review]
diff is a bit scary, but I guess it is necessary

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**